mobile/lib/modules/login/providers/authentication.provider.dart

```diff
@@ -136,38 +136,67 @@ class AuthenticationNotifier extends StateNotifier<AuthenticationState> {
   Future<bool> setSuccessLoginInfo({
     required String accessToken,
     required String serverUrl,
+    bool offlineLogin = false,
   }) async {
     _apiService.setAccessToken(accessToken);
-    UserResponseDto? userResponseDto;
-    try {
-      userResponseDto = await _apiService.userApi.getMyUserInfo();
-    } on ApiException catch (e) {
-      if (e.innerException is SocketException) {
-        state = state.copyWith(isAuthenticated: true);
+
+    // Get the deviceid from the store if it exists, otherwise generate a new one
+    String deviceId =
+        Store.tryGet(StoreKey.deviceId) ?? await FlutterUdid.consistentUdid;
+
+    bool shouldChangePassword = false;
+    User? user;
+
+    bool retResult = false;
+
+    if (offlineLogin) {
+      User? offlineUser = Store.tryGet(StoreKey.currentUser);
```

Comment:
Consider adding error handling for the case when the user is offline and the stored user data is null. This could prevent potential crashes or undefined behavior. [important]

Reply:
Good catch, fixed